resources/deb/etc/init.d/openhab2

```diff
@@ -71,15 +71,36 @@ fi
 # Load the VERBOSE setting and other rcS variables
 . /lib/init/vars.sh
 
+# Check to see if openHAB is currently running
+EXISTINGPID=`pgrep java -G openhab | tail -1`
+
 case "$1" in
   start)
         log_daemon_msg "Starting $DESC" "$NAME"
-        #${OPENHAB_INST_DIR}/bin/setpermissions.sh > /dev/null
-        if start-stop-daemon --start --quiet --pidfile $PIDFILE -c "${USER_AND_GROUP}" --umask 002 --background --make-pidfile --chdir ${OPENHAB_DIR} \
-                --oknodo --exec /usr/share/openhab2/runtime/bin/start
+
+        # If a process still exists but there's no PID file, then openHAB is probably still shutting down. We should proceed.
+        if [ -z "$EXISTINGPID" ] || [ ! -f $PIDFILE ];
         then
-            log_end_msg 0
+            if start-stop-daemon --start --quiet --pidfile $PIDFILE -c "${USER_AND_GROUP}" --umask 002 --background --chdir ${OPENHAB_DIR} \
+                    --oknodo --exec /usr/share/openhab2/runtime/bin/start
+            then
+                timeout=0
+                # The correct process is only available once the wrapper scripts have finished, we should prevent the user from running again
+                until [ $timeout -eq 5 ] || [ ! -z "$pid"  ];
```

Comment:
I added a timeout of 5 seconds which should be long enough for the wrapper scripts to finish and to start the main Java process. Unfortuately make-pidfile seems to point to a temporary process and I found many resources that suggest making your own in this case.

Furthermore, the systemd version doesn't even make a pid file,  should we not worry about it here either? The service doesn't use it in stopping itself and were only adding it here for compatibility with other programs that may use pids.